mono/utils/mono-errno.h

```diff
@@ -0,0 +1,21 @@
+/**
+ * \file
+ * Access the native error code
+ *
+ * Author:
+ *   Alexander Kyte (alkyte@microsoft.com)
+ *
+ * (C) 2018 Microsoft, Inc.
+ *
+ */
+
+#ifndef __MONO_ERRNO_H__
+#define __MONO_ERRNO_H__
+
+// Enough indirection to do something else here, or log
+
+#define SET_ERRNO(errno_val) do { \
```

Comment:
(nit) Could you make the macro definition not include a `;` and make all the callers write `SET_ERRNO (whatever);` (with a trailing `;`)?

Reply:
Or `#define SET_ERRNO(x) (errno = (x))`?

Reply:
Or an inline function, or even a function? All should be ok. errno access is already big and slow (a function call), a function won't be much bigger and slower. Inline function would be no change.